src/webui/webapplication.cpp

```diff
@@ -318,7 +342,13 @@ void WebApplication::action_command_download()
             if ((url.size() == 40 && !url.contains(QRegExp("[^0-9A-Fa-f]")))
                 || (url.size() == 32 && !url.contains(QRegExp("[^2-7A-Za-z]"))))
                 url = "magnet:?xt=urn:btih:" + url;
-            BitTorrent::Session::instance()->addTorrent(url);
+
+            BitTorrent::AddTorrentParams params;
+            params.savePath = savepath;
+            params.label = label;
+            const QUrl qurl = QUrl::fromEncoded(url.toUtf8());
```

Comment:
I was tempted to do that but I checked the code style in downloadmanager.cpp in the function DownloadManager::downloadUrl where they did just that

```    
    // Process download request
    qDebug("url is %s", qPrintable(url));
    const QUrl qurl = QUrl::fromEncoded(url.toUtf8());
    QNetworkRequest request(qurl);
```

I'll fix it since I actually prefer not creating the qurl variable too.